firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -190,7 +190,18 @@ class TrackingProtectionViewController: UIViewController,
     // MARK: Redux
     func newState(state: TrackingProtectionState) {
         trackingProtectionState = state
-        // TODO: FXIOS-10657 connect the ETP navigation with the BVC general navigation
+        if let navDestination = state.navigateTo {
+            switch navDestination {
+            case .home:
+                navigationController?.popToRootViewController(animated: true)
+            case .back:
+                navigationController?.popViewController(animated: true)
+            case .close:
+                enhancedTrackingProtectionMenuDelegate?.didFinish()
```

Comment:
these are simple calls so okay with leaving it here, but curious why not add the navigation logic to the coordinator instead?

Reply:
I'm not a big fan of one liner methods called once